### PR TITLE
Added handling of modifiers getting forwarded from FFMPEG to drm API

### PR DESCRIPTION
**Pull request by szwenni**
* changed drmModeAddFB2 to drmModeAddFB2WithModifiers
* set Modifiers flag in API call
* fetched and set modifiers accordingly to kernel constraints

As already discussed in #7492 here are the change to handle modifiers.

### CODE REVIEW

**Comment by wm4** [comment]
Thanks. Someone should update the commit message to project conventions, maybe (or maybe not) split it into multiple commits, and check whether the new drm function requires bumping the drm pkg-config check.

**Comment by xantoz** [comment]
looks generally ok to me

File: video/out/drm_prime.c [hunk 1]
```diff
@@ -39,6 +41,11 @@ int drm_prime_create_framebuffer(struct mp_log *log, int fd, AVDRMFrameDescripto
                 mp_err(log, "Failed to retrieve the Prime Handle from handle %d (%d).\n", object, descriptor->objects[object].fd);
                 goto fail;
             }
+            if(object == 0 && descriptor->objects[object].format_modifier) {
+                modifiers[object] = descriptor->objects[object].format_modifier;
+            } else if (object == 0) {
+                modifiers[object] = 0;
+            }
```
Comment (sfan5) [inline video/out/drm_prime.c]: Can't this be shortened to
```c
    if (object == 0) {
        modifiers[object] = descriptor->objects[object].format_modifier;
    }
```
?

Comment (szwenni) [reply video/out/drm_prime.c]: Should be okay

**Comment by xantoz** [comment]
merged as 8b24510

**Pull request closed without merging**